docs/Learning-Environment-Design-Agents.md

```diff
@@ -507,6 +509,49 @@ setting the State Size.
 - Use as few rays and tags as necessary to solve the problem in order to improve
   learning stability and agent performance.
 
+### Variable Length Observations
+
+It is possible to have observations be of variable size by using a `BufferSensor`,
+You can add a `BufferSensor` to your Agent by adding a `BufferSensorComponent` to
+its GameObject.
+The `BufferSensor` can be useful in situations in which the Agent must pay
+attention to a varying number of entities. On the trainer side, the `BufferSensor`
```

Comment:
Should we add an example e.g. varying number of tiles in the Sorter

Reply:
I think "variable number of enemies" or "variable number of projectiles" is a better example.